lib/ansiblelint/runner.py

```diff
@@ -100,6 +115,27 @@ def run(self) -> List[MatchError]:
                 "Examining %s of type %s",
                 ansiblelint.file_utils.normpath(file['path']),
                 file['type'])
+
+            # we should bother checking playbooks only if they pass Ansible syntax-check
+            if file['type'] == 'playbook':
+                result = subprocess.run(
+                    ['ansible-playbook', '--syntax-check', file['path']],
+                    stdin=subprocess.PIPE,
+                    stdout=subprocess.PIPE,
+                    stderr=subprocess.PIPE,
+                    shell=False,  # needed when command is a list
+                    universal_newlines=True,
+                    check=False
+                )
```

Comment:
I think that it returns at most one error is an implementation detail (which could change), so keeping this code here flexible to allow multiple errors is preferrable IMO.

Reply:
As the code stands now, it is confusing and smells like premature optimization. Since this is a private API, breaking it is not a big deal once we have a use case for that. And we also gain a meaningful commit that explains why we changed from single to multiple returns.

We can also add a comment describing how we are preparing the function for future use. Still, I am pretty sure Ansible will not report multiple errors any time soon (continuing parsing after the first error is discovered can be quite hard and not worth the complications in scenarios such as Ansible).

Reply:
@tadeboro I am still working on refactoring the Rule class methods to introduce type checks. Two more methods to go. I will do these before introducing this new rule.

I will first finish adoption of the Lintable object use before rebasing this change, we still have few places where we pass the old `(filename, type)` tuple.

Please keep posting feedback as this helps me with preparatory work.